db/migrate/20200428205535_uniqueness_rule_references_rule_index.rb

```diff
@@ -0,0 +1,28 @@
+class UniquenessRuleReferencesRuleIndex < ActiveRecord::Migration[5.2]
+  class NewRuleReferencesRule < ApplicationRecord; end
+
+  def change
+    ids_to_keep = RuleReferencesRule.group(
+      'rule_id, rule_reference_id'
+    ).having("count(*) > 1").pluck(Arel.sql('MIN(id)'))
```

Comment:
This is a nice way to find unique records :+1: If you use `select` instead of `pluck`, we will never load the ID records into Ruby memory.

```suggestion
    ).having("count(*) > 1").select(Arel.sql('MIN(id)'))
```